src/System.Collections.Immutable/tests/ImmutableArrayTest.cs

```diff
@@ -292,22 +292,22 @@ public void CovarianceImplicit()
         }
 
         [Fact]
-        public void CreateByCovariantStaticCast()
+        public void CastUpReference()
         {
             ImmutableArray<string> derivedImmutable = ImmutableArray.Create("a", "b", "c");
-            ImmutableArray<object> baseImmutable = ImmutableArray.Create<object, string>(derivedImmutable);
+            ImmutableArray<object> baseImmutable = ImmutableArray<object>.CastUp(derivedImmutable);
```

Comment:
They don't appear to have internals access but Equality checks just do reference comparisons on the underlying array.